src/main/ruby/core/io.rb

```diff
@@ -1792,7 +1788,7 @@ def lineno
   def lineno=(line_number)
     ensure_open
     raise TypeError if line_number.nil?
-    @lineno = Truffle::Fixnum.lower Integer(line_number)
+    @lineno = Integer(line_number)
```

Comment:
You can remove the explicit `nil?` check above.

Reply:
Nice catch.